feature-libs/product-configurator/rulebased/components/group-menu/configurator-group-menu.component.service.ts

```diff
@@ -0,0 +1,112 @@
+import { Inject, Injectable, PLATFORM_ID } from '@angular/core';
+import { DOCUMENT, isPlatformBrowser } from '@angular/common';
+import { BREAKPOINT, BreakpointService } from '@spartacus/storefront';
+import { take } from 'rxjs/operators';
+
+@Injectable({ providedIn: 'root' })
+export class ConfiguratorGroupMenuService {
+  constructor(
+    @Inject(PLATFORM_ID) protected platformId: any,
```

Comment:
Is this check _really_ needed in every method? `isPlatformBrowser(this.platformId)`? I hope not. That would simplify a bit the methods.

I hope we're safe in SSR, when you use `this.document` (Angular Injectable `DOCUMENT` wrapper) instead of the global `document` object.

Reply:
As far as I understand isPlatformBrowser(this.platformId) is needed in case of the server-side rendering. By navigating into the subgroups, the group menu will be rerendered. The same happens when one navigates from the subgroup to the parent group.

Reply:
I am not sure that I understand what you mean here:

> I hope we're safe in SSR, when you use this.document (Angular Injectable DOCUMENT wrapper) instead of the global document object.

Reply:
SSR breaks only when you're using globals `document` or `window` - only then you need a wrapper check `isPlatformBrowser`. (please note in the code of this class we're referencing now the global `document`, but not `this.document`).

If you're using Angular DOCUMENT object, SSR won't crash.

If you need to use global window, I recommend WindowRef.nativeWindow - wrapper done by Spartacus. It doesn't crash in SSR, but just return undefined.